packages/libsemanage/0001-remove-bzip2-dependency.patch

```diff
@@ -0,0 +1,268 @@
+From 227391a3403dc480c31e5d8dc96aaa085e91610a Mon Sep 17 00:00:00 2001
+From: Ben Cressey <bcressey@amazon.com>
+Date: Sat, 30 Mar 2024 04:57:02 +0000
+Subject: [PATCH] remove bzip2 dependency
+
+Signed-off-by: Ben Cressey <bcressey@amazon.com>
+---
+ src/Makefile          |   2 +-
+ src/compressed_file.c | 179 ++++++------------------------------------
+ src/libsemanage.pc.in |   1 -
+ tests/Makefile        |   2 +-
+ 4 files changed, 26 insertions(+), 158 deletions(-)
+
+diff --git a/src/Makefile b/src/Makefile
+index d525996..314d064 100644
+--- a/src/Makefile
++++ b/src/Makefile
+@@ -90,7 +90,7 @@ $(LIBA): $(OBJS)
+ 	$(RANLIB) $@
+ 
+ $(LIBSO): $(LOBJS)
+-	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ -lsepol -laudit -lselinux -lbz2 -Wl,-soname,$(LIBSO),--version-script=libsemanage.map,-z,defs
++	$(CC) $(CPPFLAGS) $(CFLAGS) $(LDFLAGS) -shared -o $@ $^ -lsepol -laudit -lselinux -Wl,-soname,$(LIBSO),--version-script=libsemanage.map,-z,defs
+ 	ln -sf $@ $(TARGET)
+ 
+ $(LIBPC): $(LIBPC).in ../VERSION
+diff --git a/src/compressed_file.c b/src/compressed_file.c
+index 5546b83..3632d79 100644
+--- a/src/compressed_file.c
++++ b/src/compressed_file.c
+@@ -27,148 +27,14 @@
+ #include <unistd.h>
+ #include <fcntl.h>
+ 
+-#include <bzlib.h>
```

Comment:
I wonder if upstream will be interested in a patch that includes conditionals for compressed modules. Are you aware of any discussion about other compression libraries?

Reply:
I haven't seen any recent discussion; in [past threads](https://lore.kernel.org/selinux/?q=libsemanage+compression), compression tended to be challenged with "is this really necessary?" which strikes me as totally correct and fair.

If we really needed compression, we could work around this in Bottlerocket with a combination of squashfs/erofs (as we do with licenses and kernel devel files) and overlayfs (to add a writable upper layer). But that seems like overkill here.